File: output_to_file_and_read_lines/output_to_file_and_read_lines.rb
output_file = "/tmp/output_ruby.txt"
begin
  File.open(output_file, "w") do |wfh|
    wfh.puts("Hello World!\n\nAdditional line\n")
  end
rescue SystemCallError => e
  puts "class=#{e.class},message=#{e.message}"
rescue IOError => e
  puts "class=#{e.class},message=#{e.message}"
end

line_cnt = 0
File.open(output_file, "r") do |rfh|
  while (line = rfh.gets)
    puts line
    line_cnt += 1
  end
end

puts line_cnt

File: output_to_file_and_read_lines/output_to_file_and_read_lines.py
import sys
output_file = "/tmp/output_python.txt"
try:
  with open(output_file, "w") as wfh:
    wfh.write("Hello World!\n\nAdditional line\n")
except IOError:
  print("Failed to write to " + output_file)
  sys.exit(1)
except:
  print("Unexpected error:", sys.exc_info()[0])
  sys.exit(1)

line_cnt = 0
with open(output_file, "r") as rfh:
  for line in rfh:
    print(line.rstrip("\n"))
    line_cnt += 1
print(line_cnt)
